lib/datasource/docker/index.ts

```diff
@@ -75,10 +76,21 @@ async function getDockerApiTags(
   }
   let page = 1;
   do {
-    const res = await http.getJson<{ tags: string[] }>(url, {
-      headers,
-      noAuth: true,
-    });
+    let res;
+    try {
+      res = await http.getJson<{ tags: string[] }>(url, {
+        headers,
+        noAuth: true,
+      });
+    } catch (err) {
+      if (isECRMaxResultsError(err)) {
+        const maxResults = 1000;
```

Comment:
Should we check if we already reduced page size and it fails again because of some wrong server configuration?

Reply:
To clarify, is the concern here is that we may in some special scenario get stuck processing this condition over and over?

If so, what's your preferred strategy for addressing it:
1. Setting a flag when we get this error and use it as a sentinel here to allow/deny this behavior
2. Only permit this behavior on the first iteration (`page == 1`) -- may lose out on other tags in subsequent pages
3. Another option

Reply:
I think we need to make sure we don't get into an infinite loop due to the "continue" use

Reply:
Yes, that's my concern too. We should do this only on first page.

Reply:
I was worried that it could get into an infinite loop on page 1 too